integration-tests/test_compliance.py

```diff
@@ -0,0 +1,92 @@
+    pytest.mark.skipif(
+        conftest.read_os_release() != "rhel",
+        reason="Skipping tests because OS ID is not RHEL",
+        allow_module_level=True,
+    ),
+]
+
+
+def test_compliance_option(insights_client):
+    """
+    :id: caa8b3e1-9347-494c-a1f5-1fa670136834
+    :title: Test compliance option
+    :reference: https://issues.redhat.com/browse/CCT-1229
+    :description:
+        This test verifies that running the --compliance will not result in a failure
+    :tags: Tier 1
+    :steps:
+        1. Run insights-client with --compliance option
+        2. Register insights-client
+        3. Run insights-client with --compliance option
+    :expectedresults:
+        1. Command will fail with instructions for user to register
+        2. System is successfully registered
+        3. The output of the command informs user that
+            system is not associated with any policies
+    """
+    compliance_before_registration = insights_client.run("--compliance", check=False)
+    assert compliance_before_registration.returncode == 1
+    assert (
+        "This host has not been registered. Use --register to register this host"
+        in compliance_before_registration.stdout
+    )
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    compliance_after_registration = insights_client.run("--compliance", check=False)
+    if compliance_after_registration.returncode == 1:
+        assert (
+            "System is not associated with any policies."
+            in compliance_after_registration.stdout
+        )
+    else:
+        assert "Successfully uploaded report" in compliance_after_registration.stdout
+
+
+def test_compliance_policies_option(insights_client):
+    """
+    :id: ad3a2073-3a2e-485e-bc7b-fede2111a06a
+    :title: Test compliance-policies option
+    :reference: https://issues.redhat.com/browse/CCT-1229
+    :description:
+        This test verifies that running the --compliance-policies
+        will not result in a failure
+    :tags: Tier 1
+    :steps:
+        1. Register insights-client
+        2. Run insights-client with --compliance-policies option
+    :expectedresults:
+        1. System is successfully registered
+        2. The output of the command informs user that
+            system is not assignable to any policy
+    """
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    compliance_policies = insights_client.run("--compliance-policies", check=False)
+    if compliance_policies.returncode == 1:
+        assert "System is not assignable to any policy." in compliance_policies.stdout
+        assert (
+            "Create supported policy using the Compliance web UI."
+            in compliance_policies.stdout
+        )
+    else:
+        # Checking only if some policy is available - ID will be available if yes
+        assert "ID" in compliance_policies.stdout
```

Comment:
I would rather see this comment in the `:expectedresults:` in the function docstring.
Also rephrase it a little. Something like: `Verify that at least one policy is available - if so, ID must be present`